client/state/data-layer/wpcom/account-recovery/index.js

```diff
@@ -0,0 +1,21 @@
+/**
+ * Internal dependencies
+ */
+import wpcom from 'lib/wp';
+
+import {
+	ACCOUNT_RECOVERY_RESET_OPTIONS_RECEIVE,
+	ACCOUNT_RECOVERY_RESET_OPTIONS_ERROR,
+} from 'state/action-types';
+
+export const requestResetOptions = ( { dispatch }, { userData } ) => (
+	wpcom.undocumented().accountRecoveryReset( userData ).getResetOptions()
```

Comment:
Has this been in `undocumented()` for a while? It's fine here, but ideally we'd start moving code out of `wpcom.js` and into this middleware _if_ it's code that is more specific to Calypso than generally to the WordPress.com API. In any case, when adding new code, it's viable to use `wpcom.req.get( … )` directly in this middleware layer because hopefully soon we will have a functioning HTTP layer to handle the actual requests, error-handling, and optimization for us.